libraries/core-react/src/TableOfContents/TableOfContents.tsx

```diff
@@ -19,7 +26,7 @@ const StyledTableOfContents = styled.nav`
     `}
 `
 
-const TocList = styled((props) => <List {...props} />)`
+const TocList = styled((props: ListType) => <List {...props} />)`
```

Comment:
This has nothing to do with the typescript rewrite, more clean code. But these should be normal "extends". 

For examples; 
`const TocList = styled((props: ListType) => <List {...props} />)` 
is equivalent 
 `const TocList = styled(List)`